server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTasksLogger.java

```diff
@@ -36,30 +38,37 @@ public class TopNSearchTasksLogger implements Consumer<Task> {
     private static final Logger SEARCH_TASK_DETAILS_LOGGER = LogManager.getLogger(TASK_DETAILS_LOG_PREFIX + ".search");
 
     // number of memory expensive search tasks that are logged
-    private static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
+    public static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
         LOG_TOP_QUERIES_SIZE,
         10,
+        1,
+        25,
```

Comment:
How did we arrive at this max threshold?

Reply:
Sruti has already done testing earlier so after discussion with her, we decided these thresholds. Increasing top n queries size or increasing frequency may affect the performance, that's why we have these thresholds.

Reply:
Can you add links/references for the testing done? I understand there should be a max cap to avoid verbose logging, but is 25 the right number?